addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -235,6 +245,14 @@ function factory(dependencies) {
             return clear();
         }
 
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeHasLinkPreview() {
+            return this.mimetype === 'application/o-linkpreview';
```

Comment:
Avoid this, it would probably be acceptable for now to have a condition on the attachment having both a url and data, and of which data is an image.